tracer/test/Datadog.Trace.Tests/DatabaseMonitoring/DatabaseMonitoringPropagatorTests.cs

```diff
@@ -115,10 +117,55 @@ public void ExpectedCommentInjectedV1()
             span.SetTraceSamplingPriority(samplingPriority);
 
-            var returnedComment = DatabaseMonitoringPropagator.PropagateSpanData(dbmPropagationLevel, "Test.Service", "MyDatabase", "MyHost", span, integrationId, out var traceParentInjectedValue);
+            var returnedComment = DatabaseMonitoringPropagator.PropagateDataViaComment(dbmPropagationLevel, "Test.Service", "MyDatabase", "MyHost", span, integrationId, out var traceParentInjectedValue);
 
             traceParentInjectedValue.Should().Be(traceParentInjected);
             returnedComment.Should().Be(expectedComment);
         }
+
+        [Theory]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserKeep, true, "01000000000000BEEF0000000000000000000000000000CAFE")]
+        [InlineData("full", "sqlclient", SamplingPriorityValues.UserReject, true, "00000000000000BEEF0000000000000000000000000000CAFE")]
+        [InlineData("nope", "sqlclient", SamplingPriorityValues.UserKeep, false, null)]
+        // disabled for all db types except mysql for now
+        [InlineData("full", "npgsql", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "sqlite", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "oracle", SamplingPriorityValues.UserKeep, false, null)]
+        [InlineData("full", "mysql", SamplingPriorityValues.UserKeep, false, null)]
+        public void ExpectedContextSet(string propagationMode, string integration, int? samplingPriority, bool shouldInject, string expectedContext)
+        {
+            Enum.TryParse(propagationMode, ignoreCase: true, out DbmPropagationLevel dbmPropagationLevel);
+            Enum.TryParse(integration, ignoreCase: true, out IntegrationId integrationId);
+
+            // capture command and parameter sent
+            string sql = null;
+            byte[] context = null;
+            var connectionMock = new Mock<IDbConnection>(MockBehavior.Strict);
+            var commandMock = new Mock<IDbCommand>();
+            var parameterMock = new Mock<IDbDataParameter>();
+            connectionMock.Setup(c => c.CreateCommand()).Returns(commandMock.Object);
+            commandMock.SetupSet(c => c.CommandText = It.IsAny<string>())
+                       .Callback<string>(value => sql = value);
+            commandMock.Setup(c => c.CreateParameter()).Returns(parameterMock.Object);
+            commandMock.SetupGet(c => c.Parameters).Returns(Mock.Of<IDataParameterCollection>());
+            parameterMock.SetupSet(p => p.Value = It.IsAny<byte[]>())
+                         .Callback<object>(value => context = (byte[])value);
+
+            var span = _v0Tracer.StartSpan("mysql.query", parent: SpanContext.None, serviceName: "pouet", traceId: (TraceId)0xCAFE, spanId: 0xBEEF);
```

Comment:
`mysql.query`?

Reply:
we do the same above though, a span called `mysql` even though we test for every DB system
https://github.com/DataDog/dd-trace-dotnet/blob/c87e6d2f839e9d5f9aed707aa1e1be7febbdebaa/tracer/test/Datadog.Trace.Tests/DatabaseMonitoring/DatabaseMonitoringPropagatorTests.cs#L59

Reply:
Oh, I see. If the operation name is not relevant to the test (not verified), maybe use something generic here like `db.query` to avoid confusion? Low priority.